core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -161,6 +165,34 @@ static ClientRequestContextBuilder builder(RpcRequest request, URI uri) {
         return new ClientRequestContextBuilder(request, uri);
     }
 
+    /**
+     * Returns the {@link ServiceRequestContext} whose {@link Service} invokes the {@link Client}
+     * {@link Request} which created this {@link ClientRequestContext}.
+     */
+    @Nullable
+    ServiceRequestContext serviceRequestContext();
+
+    /**
+     * {@inheritDoc} If {@link #serviceRequestContext()} is not {@code null} and the
+     * {@link ServiceRequestContext} has the {@link Attribute}, this will return the {@link Attribute}.
+     */
+    @Override
```

Comment:
`@Nullable` I think

Reply:
I think it's not nullable because we always make an empty `DefaultAttribute`